source/extensions/filters/http/ip_tagging/ip_tagging_filter.cc

```diff
@@ -83,6 +83,11 @@ Http::FilterHeadersStatus IpTaggingFilter::decodeHeaders(Http::RequestHeaderMap&
   if (!tags.empty()) {
     const std::string tags_join = absl::StrJoin(tags, ",");
     headers.appendEnvoyIpTags(tags_join, ",");
+    if (!config_->customHeader().get().empty()) {
+      // remove the header if it already exists and then set it with the tags.
+      headers.remove(config_->customHeader());
+      headers.addReferenceKey(config_->customHeader(), tags_join);
+    }
```

Comment:
> seems if the custom header is set, both the custom header and envoy ip tags will be set? Is this expected?
> 
> tips: you can use `setReferenceKey` to override exist header directly.

yes I think so. Seems the default x-envoy-ip-tags header is only available within the envoy, if we want to send this info to upstream service, we need a separate header; and there might be other envoy http filters relying on the default x-envoy-ip-tags header, if we don't set it, it could impact the existing filters.

Reply:
I don't see any use case of this header. If some other third party filters depends on this header, then it should not set the `custom_header`.